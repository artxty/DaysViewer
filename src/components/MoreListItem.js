import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet, Text, View } from 'react-native';

const styles = StyleSheet.create({
  item: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  text: {
    fontSize: 18,
  },
});

export default function ListItem({
  label, children, style, onPress,
}) {
  return (
    <View style={{ ...styles.item, ...style }} onStartShouldSetResponder={onPress}>
      <Text style={styles.text}>{label}</Text>
      {children}
    </View>
  );
}

ListItem.propTypes = {
  label: PropTypes.string.isRequired,
  children: PropTypes.element.isRequired,
  style: PropTypes.shape({}),
  onPress: PropTypes.func,
};

ListItem.defaultProps = {
  style: StyleSheet.create({}),
  onPress: () => null,
};
