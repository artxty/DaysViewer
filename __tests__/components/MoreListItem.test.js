import React from 'react';
import renderer from 'react-test-renderer';

import MoreListItem from '../../src/components/MoreListItem';

const ChildrenComponent = () => <div />;

describe('MoreListItem', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <MoreListItem label="test">
        <ChildrenComponent />
      </MoreListItem>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
