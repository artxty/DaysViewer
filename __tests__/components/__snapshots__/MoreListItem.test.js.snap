// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoreListItem renders correctly 1`] = `
<View
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 18,
      }
    }
  >
    test
  </Text>
  <div />
</View>
`;
